test-harness/scenarios/strict-options.scenario

```diff
@@ -20,8 +20,8 @@ Options:
       --version                  Show version number  [boolean]
       --help                     Show help  [boolean]
   -e, --encoding                 text encoding to use  [string] [choices: "utf8", "ascii", "utf-8", "utf16le", "ucs2", "ucs-2", "base64", "latin1"] [default: "utf8"]
-  -f, --format                   formatter to use for outputting results  [string] [choices: "json", "stylish", "junit", "html", "text", "teamcity", "pretty"] [default: "stylish"]
```

Comment:
I don't really like that we lost the `choices` in the help text.

Wondering whether it would be easier (implementation- and usage-wise) to allow multiple `--format` and `--output` parameters instead of the providing multiple values within one parameter.

Reply:
Initially, I was looking at having `--format json junit` but `yargs` [does not behave well](https://github.com/yargs/yargs/blob/main/docs/api.md#arraykey) with array arguments together with positional ones.

Repeating the `--format` flag is [completely doable](https://github.com/yargs/yargs/blob/main/docs/tricks.md#arrays) instead

Reply:
We could also consider https://github.com/yargs/yargs/blob/main/docs/tricks.md#objects to make the implementation even easier (e.g. `--output.junit`)

Reply:
yup, it feels like using objects might be a good idea for outputs.

I'm trying to explore how invasive changing the type of `format` option would be.
It feels like we'd need to consider it a breaking change if we changed the type of the option, as `yargs` would need to be instrumented with `--` to stop the parsing, or, alternatively, pass `format` after the positional arguments.

Reply:
thoughts https://github.com/stoplightio/spectral/pull/2037?